miniscope_io/stream_daq.py

```diff
@@ -347,9 +378,29 @@ def _buffer_to_frame(
                     break
 
                 header_data, serial_buffer = self._parse_header(serial_buffer)
-                serial_buffer = self._trim(
-                    serial_buffer, self.buffer_npix[header_data.frame_buffer_count], locallogs
-                )
+
+                # If metadata queue is full, pop the oldest one. Probably not the best way.
+                metadata_list.append(header_data)
```

Comment:
i think [deque](https://docs.python.org/3/library/collections.html#collections.deque)?

Reply:
Oh, my bad, that's an old comment from when this was actually a queue. I ended up thinking using a shared list (eventually circular buffer or something if size becomes a problem) might be better for metadata because there's no peek function for `mp.queue`.

Reply:
Ooooh this list is shared between processes? That might cause some problems.

Reply:
Yes, and yes, obviously, it might lead to issues, but we need something like this.
- This is for sharing metadata so other processes (e.g., visualization stuff in the main process, callbacks) can use it.
- I initially wanted to use `multiprocessing.Queue` because those are thread/process-safe. However, native methods like `get()` need to consume data, which wouldn't make sense when we want an array that multiple processes can refer to.

So, the two options I thought of were:
1. Make a shared list with locked io methods.
2. Make a wrapper of `mp.Queue` with peek methods. Probably the simplest is to `get()` all data and re-`put()` them in each method, but it still feels pretty complex and could have timing issues.

I just started off with (1) because it seemed simpler. But I don't have much confidence in this so I would appreciate your thoughts @sneakers-the-rat.

Reply:
oh! i didn't see it was a list from the multiprocessing manager. that should be less of a problem. i'm going to run this now and write some tests and i'll get back to ya